### PR TITLE
Enable java 11 features for che-server code

### PR DESCRIPTION
**Pull request by skabashnyuk**
### What does this PR do?
Enable java 11 features for che-server code

### What issues does this PR fix or reference?
n/a

#### Release Notes
n/a

#### Docs PR
n/a

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -41,8 +41,6 @@
     </scm>
     <properties>
         <?SORTPOM IGNORE?>
-        <java.target.version>8</java.target.version>
-        <jdk.min.version>11</jdk.min.version>
```
Comment (sparkoo) [inline pom.xml:45]: why not set them both to 11? Is it defined somewhere else?

Comment (skabashnyuk) [reply pom.xml:45]: yes. in the parent pom https://github.com/eclipse/che-parent/blob/master/pom.xml#L102-L103

**Comment by skabashnyuk** [comment]
[crw-ci-test]

**Comment by skabashnyuk** [comment]
[ci-build]

**Review by sparkoo — APPROVED** [review]

**Pull request merged**